state/protocol/protocol_state/epochs/fallback_statemachine_test.go

```diff
@@ -692,48 +694,54 @@ func (s *EpochFallbackStateMachineSuite) TestProcessingMultipleEventsAtTheSameBl
 	rapid.Check(s.T(), func(t *rapid.T) {
 		s.SetupTest() // start each time with clean state
 		var events []flow.ServiceEvent
+		// ATTENTION: explicitly grouped draw events because drawing an int range can raise a panic.
```

Comment:
Why can drawing an int range panic? Do you mean that for example this line can panic?
```
setupEvents := rapid.IntRange(0, 5).Draw(t, "number-of-setup-events")
```

Reply:
yes, this line can panic but `rapid.Check` has a [recover](https://go.dev/blog/defer-panic-and-recover) in their [implementation](https://github.com/flyingmutant/rapid/blob/master/engine.go#L366) so effectively it ignores a panic if it was raised in their logic that is why it is not propagated to the test itself and it consumed by `rapid.Check`.

Reply:
I understand the reasoning that, if rapid can panic when `Draw`-ing, then doing all the `Draw`s before setting expectations can avoid a test failure. 

What I'm trying to understand is why `Draw`-ing can panic in the first place. 

I see it will panic [creating a generator if the range is invalid](https://github.com/flyingmutant/rapid/blob/master/integers.go#L159), but we're passing in constant, valid ranges.